app/client/src/sagas/PostEvaluationSagas.ts

```diff
@@ -337,55 +337,36 @@ export function* postEvalActionDispatcher(actions: Array<AnyReduxAction>) {
 // is accurate
 export function* updateTernDefinitions(
   dataTree: DataTree,
-  updates?: DataTreeDiff[],
+  updates: DataTreeDiff[],
+  isCreateFirstTree: boolean,
```

Comment:
In `isCreateFirstTree=true` flow updates are sent as `undefined`. Hence, the earlier type was correctly set according to that flow. 
```suggestion
  isCreateFirstTree: boolean,
  updates?: DataTreeDiff[],
```

Reply:
The split eval & linting PR actually changed it to []. updates for createFirstTree is now an empty array and not undefined. https://github.com/appsmithorg/appsmith/blob/49a98b5048374d358c72c7b81d2253199dad1f71/app/client/src/workers/Evaluation/evaluation.worker.ts#L226